File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/dominio/calculo/ferias/regras/PrazoDeFeriasValidRule.java
/*
 * Decompiled with CFR 0.152.
 */
package br.jus.trt8.pjecalc.negocio.dominio.calculo.ferias.regras;

import br.jus.trt8.pjecalc.base.comum.validadores.ValidatorContext;
import br.jus.trt8.pjecalc.negocio.comum.validators.ValidRule;
import br.jus.trt8.pjecalc.negocio.comum.validators.ValidValueValidator;
import br.jus.trt8.pjecalc.negocio.constantes.Mensagens;

public class PrazoDeFeriasValidRule
implements ValidRule {
    @Override
    public Mensagens getMessage() {
        return Mensagens.MSG0004;
    }

    @Override
    public boolean isValid(ValidValueValidator validator, ValidatorContext context) {
        Integer value = (Integer)context.getValue();
        return value == null || value.compareTo(0) >= 0;
    }
}

File: src/lib/pjecalc/core/dominio/calculo/ferias/regras/prazo-de-ferias-valid-rule.ts
/**
 * Porte 1:1 de PrazoDeFeriasValidRule.java (24 linhas).
 *
 * Regra: prazo deve ser >= 0.
 *
 * Ref: pjecalc-fonte/.../dominio/calculo/ferias/regras/PrazoDeFeriasValidRule.java
 */
export class PrazoDeFeriasValidRule {
  static readonly MESSAGE = 'MSG0004';

  static isValid(value: number | null): boolean {
    return value === null || value >= 0;
  }
}
